instance/instance.go

```diff
@@ -334,108 +239,90 @@ func (s *Store) Update(w http.ResponseWriter, r *http.Request) {
+		//edition confirmation is a one time process - cannot be editted for an instance once done
+		if instance.State == models.EditionConfirmedState && instance.Version == 0 {
 			if instance.Edition == "" {
 				instance.Edition = currentInstance.Edition
 			}
-			edition := instance.Edition
-			editionAuditParams := common.Params{"instance_id": instanceID, "dataset_id": datasetID, "edition": edition}
-			editionLogData := audit.ToLogData(editionAuditParams)
-
-			editionDoc, auditAction, err := func() (*models.EditionUpdate, string, error) {
-				// Only create edition if it doesn't already exist
-				editionDoc, action, err := s.getEdition(ctx, datasetID, edition, instanceID)
-				if err != nil {
-					log.ErrorCtx(ctx, errors.WithMessage(err, "instance update: store.getEdition returned an error"), editionLogData)
-					return nil, action, err
-				}
-
-				// Update with any edition.next changes
-				editionDoc.Next.State = instance.State
-				if err = s.UpsertEdition(datasetID, edition, editionDoc); err != nil {
-					log.ErrorCtx(ctx, errors.WithMessage(err, "instance update: store.UpsertEdition returned an error"), editionLogData)
-					return nil, action, err
-				}
-
-				return editionDoc, action, nil
-			}()
-			if err != nil {
-				if auditAction == "" {
-					// No action to audit so return early
-					return err
-				}
-
-				if auditErr := s.Auditor.Record(ctx, auditAction, audit.Unsuccessful, editionAuditParams); auditErr != nil {
-					err = auditErr
-				}
-
-				return err
-			}
-
-			s.Auditor.Record(ctx, auditAction, audit.Successful, editionAuditParams)
-
-			log.InfoCtx(ctx, "instance update: created edition", editionLogData)
 
-			// Check whether instance has a version
-			if currentInstance.Version < 1 {
-				// Find the latest version for the dataset edition attached to this
-				// instance and append by 1 to set the version of this instance document
-				version, err := s.GetNextVersion(datasetID, edition)
-				if err != nil {
-					log.ErrorCtx(ctx, errors.WithMessage(err, "instance update: store.GetNextVersion returned an error"), logData)
-					return err
-				}
+			edition := instance.Edition
+			editionLogData := log.Data{"instance_id": instanceID, "dataset_id": datasetID, "edition": edition}
 
-				instance.Version = version
+			editionDoc, editionConfirmErr := s.confirmEdition(ctx, datasetID, edition, instanceID)
+			if editionConfirmErr != nil {
+				log.ErrorCtx(ctx, errors.WithMessage(editionConfirmErr, "instance update: store.getEdition returned an error"), editionLogData)
+				return nil, editionConfirmErr
+			}
 
-				instance = s.addInstanceLinks(ctx, instance, editionDoc)
+			//update instance with confirmed edition details
+			instance.Links = currentInstance.Links
+			instance.Links.Edition = editionDoc.Next.Links.Self
+			instance.Links.Version = editionDoc.Next.Links.LatestVersion
+			instance.Version, editionConfirmErr = strconv.Atoi(editionDoc.Next.Links.LatestVersion.ID)
```

Comment:
doesn't really matter but we seem to use `strconv.ParseInt` in some places and `strconv.Atoi` in others

Reply:
there's only one occurrence of ParseInt I can find, where we're passing it to an int64, so seems correct things in correct places to me?